src/main/java/org/broadinstitute/hellbender/engine/dataflow/datasources/VariantShard.java

```diff
@@ -0,0 +1,98 @@
+package org.broadinstitute.hellbender.engine.dataflow.datasources;
+
+import com.google.cloud.dataflow.sdk.coders.*;
+import com.google.cloud.dataflow.sdk.values.KV;
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * VariantShard is section of the genome that's used for sharding work for pairing things with
+ * variants. The primary use case is pairing reads with overlapping variants.
+ * This class designed to be simple a data-storage class with related static utilities.
+ */
+public final class VariantShard {
```

Comment:
I don't see why this class needs to exist at all. Use intervals. Otherwise it's going to be a pain managing all these extra coders when we optimize them (it's very likely that we'll have to do this at some point).